sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveNamespaceAndTable.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.connector.catalog.{CatalogManager, CatalogV2Util, LookupCatalog}
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+
+case class ResolveNamespaceAndTable(catalogManager: CatalogManager)
+  extends Rule[LogicalPlan] with LookupCatalog {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+    case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
+      ResolvedNamespace(catalog.asNamespaceCatalog, ns)
+
+    case u @ UnresolvedTable(CatalogAndIdentifier(catalog, ident)) =>
```

Comment:
My point is, we should resolve `UnresolvedTable` in the same way as `UnresolvedRelation`.

`UnresolvedRelation` is resolved by 3 rules:
1. `ResolveTempViews`: this rule converts `UnresolvedRelation` to temp views if name matches
2. `ResolveTables`: this rule resolves `UnresolvedRelation` by looking up table from v2 catalogs. If the associated catalog is not v2 catalog, this rule will be skipped.
3. `ResolveRelations`: this rule resolves `UnresolvedRelation` by looking up table from the session catalog. If the associated catalog is not session catalog, this rule will be skipped.

Reply:
The rule name might be a little confusing, we can fix it later.

Reply:
I see.
if CommentOnTable is matched by `ResolveTempViews ` or `ResolveRelations `, do we need to add Exec or Command to support?

Reply:
I separated namespace and table resolving rule, the table one can visit view first, then v2, v1.